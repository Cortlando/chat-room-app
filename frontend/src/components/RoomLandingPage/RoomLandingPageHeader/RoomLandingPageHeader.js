import React from 'react'

import { AppBar, IconButton, Typography, Toolbar } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import AddIcon from '@material-ui/icons/Add'


import CreateRoom from './CreateRoom/CreateRoom'
import LoginButton from './LoginButton/LoginButton'
import LogoutButton from './LogoutButton/LogoutButton'
import Profile from './Profile/Profile'

import { useAuth0 } from '@auth0/auth0-react'


const useStyles = makeStyles((theme) => ({
    root: {
      flexGrow: 1,
    },
    menuButton: {
      marginRight: theme.spacing(2),
    },
    title: {
      flexGrow: 1,
    },
  }));

const RoomLandingPageHeader = ({ rooms, setRooms }) => {

    const classes = useStyles();

    const [open, setOpen] = React.useState(false);

    const { isAuthenticated } = useAuth0()

    const handleOpen = () => {
        setOpen(true);
    };

    const handleClose = () => {
        setOpen(false);
    };

    console.log(rooms, setRooms)

    
    return (
        <div className={classes.root}>
          <AppBar position="static">
            <Toolbar>
              <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu" onClick={handleOpen}>
                <AddIcon />
              </IconButton>
              <Typography variant="h6" className={classes.title}>
                Click to create a room
              </Typography>
              <CreateRoom open={open} setOpen={setOpen} handleOpen={handleOpen} handleClose={handleClose} />
              {isAuthenticated ? <LogoutButton/> : <LoginButton/>}
                <Profile/>
            </Toolbar>
          </AppBar>
        </div>
      );

}



export default RoomLandingPageHeader


