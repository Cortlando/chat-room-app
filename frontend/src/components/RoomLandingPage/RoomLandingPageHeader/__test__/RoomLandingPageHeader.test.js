import {render, screen, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer'

import RoomLandingPageHeader from '../RoomLandingPageHeader'


let container = null
afterEach(() =>{
    cleanup();
})

beforeEach(() =>{
    container = document.createElement("div");
    document.body.appendChild(container)
})

test('should display "Click to create a room(Refresh to see rooms)"', () => {
    render(<RoomLandingPageHeader/>)

    let element = screen.getByTestId(`Tid1`)

    //console.log(element)
    expect(element).toBeInTheDocument();
    expect(element).toHaveTextContent("Click to create a room(Refresh to see rooms)")
   // expect(element).toContainHTML('<h6>')
    
})

test('matches snapshot', ()=>{
    const tree = renderer.create(<RoomLandingPageHeader/>).toJSON();
    expect(tree).toMatchSnapshot();

})

