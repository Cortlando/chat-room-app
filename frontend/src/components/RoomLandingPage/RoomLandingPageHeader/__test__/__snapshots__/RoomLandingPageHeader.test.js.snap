// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="makeStyles-root-5"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-6 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
      </button>
      <h6
        className="MuiTypography-root makeStyles-title-7 MuiTypography-h6"
        data-testid="Tid1"
      >
        Click to create a room(Refresh to see rooms)
      </h6>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Log In
        </span>
      </button>
      <div>
        Loading ...
      </div>
    </div>
  </header>
</div>
`;
